server/serverimpl.go

```diff
@@ -159,7 +169,11 @@ func (s *server) Stop(ctx context.Context) error {
 		defer func() { s.httpServer = nil }()
 		// This stops accepting new connections. TODO: close existing
 		// connections and wait them to be terminated.
-		return s.httpServer.Shutdown(ctx)
+		err := s.httpServer.Shutdown(ctx)
+		if err != nil {
+			return err
+		}
+		s.httpServerServeWg.Wait()
```

Comment:
This needs to honour `ctx` cancellation. If context is Done(), Stop() needs to return with an error. I am not sure if this is going to be the behavior, given that Wait() can wait indefinitely long. Please add a test that verifies the cancellation case.

Reply:
Makes sense to me, added in https://github.com/open-telemetry/opamp-go/pull/348/commits/4e4986f743b387b30c1b3725125a22db18369c51